pkg/auth/authenticator/password/ldappassword/ldap.go

```diff
@@ -88,18 +83,15 @@ func (a *Authenticator) getIdentity(username, password string) (authapi.UserIden
 		return nil, false, nil
 	}
 
-	// Make the connection
+	// Make the connection and bind to it if a bind DN and password were given
 	l, err := a.options.ClientConfig.Connect()
 	if err != nil {
 		return nil, false, err
 	}
 	defer l.Close()
 
-	// If specified, bind the username/password for search phase
-	if len(a.options.BindDN) > 0 {
-		if err := l.Bind(a.options.BindDN, a.options.BindPassword); err != nil {
-			return nil, false, err
-		}
+	if _, err := a.options.ClientConfig.Bind(l); err != nil {
```

Comment:
Bind is just one of many LDAP operations you can do (and can do repeatedly, with different dn/passwords). It is not inherent to establishing a connection. The fact that we call bind again, differently, 50 lines later in this file illustrates why bindDN/bindPassword aren't really part of the connection operation, and probably shouldn't be part of the client options.